ansible_rulebook/engine.py

```diff
@@ -217,14 +221,45 @@ async def start_source(
         )
 
 
+class RulebookFileChangeHandler(FileSystemEventHandler):
+    modified = False
+
+    def on_modified(self, event):
+        logger.debug(f"Rulebook file {event.src_path} has been modified")
+        self.modified = True
+
+    def is_modified(self):
+        return self.modified
+
+
+async def monitor_rulebook(rulebook_file):
+    event_handler = RulebookFileChangeHandler()
+    to_observe = os.path.abspath(rulebook_file)
+    observer = Observer()
+    observer.schedule(event_handler, to_observe, recursive=True)
```

Comment:
I think `recursive` flag is not necessary, am I right?

Reply:
forgot to remove it as I was trying it in different places earlier 😅  thanks for spotting it!

Reply:
actually, no.
It seems to be required to capture the modification of itself, as I've just tried omitting the `recursive=True` and without it, does not trigger the `RulebookFileChangeHandler`

Reply:
Interesting, I did a couple of tests and for me it worked, maybe it depends on the platform, not sure.